### PR TITLE
adds ability to save simple search as view

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/22452

## Testing
**URL to test:** 

https://deploy-preview-1595--atd-moped-main.netlify.app/

**Steps to test:**
- go to project list view and execute a simple search
- the "save view" button should appear, click on it and note the description field is autopopulated with the search term
- save the view and confirm it appears in the saved views table and the link works
- go to project list view and execute a simple search, then add a filter (or vice versa)
- the "save view" button should appear, click on it and note the description field is autopopulated with the search term plus the filters applied
- save the view and confirm it appears in the saved views table and the link works

try out different combinations, adding a simple search and filters in different orders, removing one or the other, and saving the view. all combinations and orders should work!

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view)
- [x] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable
   - Added new test, "Now, filter using only the simple search and make sure the "Save view" button appears so you can save a simple search term with or without advanced filters."

### CODE REVIEW

File: moped-editor/src/components/GridTable/SearchBar.js [hunk 1]
```diff
@@ -165,7 +166,7 @@ const SearchBar = ({
         variant="outlined"
         value={searchFieldValue}
       />
-      {filterStateActive && !advancedSearchAnchor && (
+      {(filterStateActive || searchTerm) && !advancedSearchAnchor && (
```
Comment (tillyw) [inline moped-editor/src/components/GridTable/SearchBar.js:169]: this works fine since there will be no filters to display if the filter state is not active. it may be worth considering breaking out the 'save view' button into its own component separate from the `<FiltersChips>` but I wasn't sure if that would fit the scope of this issue

File: moped-editor/src/views/projects/projectsListView/components/SaveUserViewModal.js [hunk 2]
```diff
@@ -13,17 +13,32 @@ const SaveUserViewModal = ({
...
 
   let { pathname, search } = useLocation();
 
-  const defaultDescription = filtersLabels
-    .map(
-      (filter) =>
-        `${filter.filterLabel} ${filter.operatorLabel} ${filter.filterValue}`
-    )
-    .join(", ");
+  const getDefaultDescription = () => {
```
Comment (tillyw) [inline moped-editor/src/views/projects/projectsListView/components/SaveUserViewModal.js:22]: sets the default description based on what the user has selected

Comment (mddilley) [reply moped-editor/src/views/projects/projectsListView/components/SaveUserViewModal.js:22]: @tillyw I think this could be a good place to use `useMemo`.

Maybe something like:
```js
const defaultDescription = useMemo(() => {
// all the same default name prep in your function
}, [the dependencies here])
```

**Review by frankhereford — APPROVED** [review]
This looks great Tilly - thank you for this. I have reviewed your code, as well as tested the behavior of the button, modal and search bar and how they interplay with each other. I think this is ready to ship. Thanks again!

**Comment by roseeichelmann** [comment]
image

This is the description i got in the dashboard when i typed in "test" as a simple search and saved "test" as the description in the modal?

**Review by chiaberry — COMMENTED** [review]
I found an edge case where if I put something in the simple search, save the view and then select the search term and enter a new search term, the "save view" button is still disabled. In other words, I searched a new term without fully deleting the first one, just replacing it.

**Review by mddilley — CHANGES_REQUESTED** [review]
This is working well, and it is nice to be able to save the simple search on its own! I requested a couple of small changes.

**Comment by johnclary** [comment]
> I found an edge case where if I put something in the simple search, save the view and then select the search term and enter a new search term, the "save view" button is still disabled.

The save view button is re-enabled for me once I modify the search term and press the **Enter** key, which seems consistent since one must press **Enter** in order to initiate text search anyway 🤔

File: moped-editor/src/components/GridTable/Search.js [hunk 3]
```diff
@@ -184,6 +184,7 @@ const Search = ({
                 filtersConfig={filtersConfig}
                 resetSimpleSearch={resetSimpleSearch}
                 setIsOr={setIsOr}
+                searchTerm={searchTerm}
```
Comment (johnclary) [inline moped-editor/src/components/GridTable/Search.js:187]: I noticed that leading/trailing whitespace are preserved in the search term. Might be nice to trim that when the user executes the search?

**Review by johnclary — APPROVED** [review]
Looks + works great. Thank you for enabling saving of simple searches!

🚢 🚢 🚢 🚢 

Not related to changes in this PR, I wanted to mention the tiniest bit of awkward behavior when modifying the search input at the same time as modifying the advanced filters.

If I do a simple search, then modify my search input, then add an advanced filter and press **Apply**, the modified search input is not included in the search, and it's not included when I save the view.

I also noticed that If I start modifying the advanced search filters and then, with the filter menu still open, I modify the simple search input, the filter menu closes and i lose those changes. It would be nice if you could use the simple search and advanced filters in tandem and expect the **Apply** button to apply the updated search input and the updated advanced filters.

**Review by chiaberry — APPROVED** [review]
looks great! thanks for fixing the edge case

File: moped-editor/src/components/GridTable/Search.js (hunk 3 above)
Comment (mddilley) [reply moped-editor/src/components/GridTable/Search.js:187]: @tillyw could you sneak this change to trim in before merge? i noticed that the extra whitespace can make its way into the saved view descriptions as well 🙏

Screenshot 2025-05-21 at 11 50 42 AM

**Review by mddilley — APPROVED** [review]
Looks and works so great! Thanks for updating to re-enable the button! 🚢 🚀

**Comment by mddilley** [comment]
@mateoclarke I've also been seeing that error on branches that don't have your metadata update in them yet since the `./hasura-cluster setenv local` command was required to have a `config.yml` at all. Now that you git-tracked it we should see this error gone completely! 🙏

I'm also glad to see you expecting that behavior for the saved view button. Tilly had the same feedback, and I'm expecting to hear that from users if we make the saved views more and more accessible like in the ACC designs. I think it would be a really nice improvement! https://github.com/cityofaustin/atd-data-tech/issues/22411
